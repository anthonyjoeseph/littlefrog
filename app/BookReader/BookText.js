// app/BookReader/BookText.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  ScrollView,
  Animated
} from 'react-native';

class BookText extends Component{
  constructor(props){
    super(props);
    this.state = {
      translateAnim: new Animated.Value(0),
      isVisible: false
    };
  }

  componentDidMount(){
    this.slide();
  }

  slide(){
    var newFinalValue
    if(this.state.isVisible){
      newFinalValue = 0;
      this.state.isVisible = false;
    }else{
      newFinalValue = 1;
      this.state.isVisible = true;
    }
    Animated.timing(
      this.state.translateAnim,
      {
        toValue: newFinalValue,
        duration: 1000
      }
    ).start();
  }

  render(){
      return (
        <Animated.View style={[this.props.style,
            {
              backgroundColor: '#1586BA',
              transform:
                [{
                  translateX: this.state.translateAnim.interpolate({
                    inputRange: [0, 1],
                    outputRange: [-this.props.style.width, 0]
                  })
                }]
            }
            ]
          } >
          <ScrollView>
            <Text style={{color:'white'}}>{this.props.caption}</Text>
          </ScrollView>
        </Animated.View>
      );
  }
}

export default BookText;
