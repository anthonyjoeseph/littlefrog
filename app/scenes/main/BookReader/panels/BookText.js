// @flow
// app/BookReader/BookText.js

import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  ScrollView,
  Animated,
  Props,
  State
} from 'react-native';

import SubPanel from './SubPanel'

class BookText extends Component{
  state:State
  constructor(props:Props){
    super(props);
    this.state = {
      textForPages: {0:"retrieving book text..."}
    };
  }

  componentWillReceiveProps(props:Props){
    if(props.authHeaders.hasOwnProperty('Authorization')){
      fetch(props.baseURI + "text/BookTextEng.json", {
        method: "GET",
        headers: props.authHeaders
      })
      .then(function(rawBookTextData){
        return rawBookTextData.json();
      })
      .then(function(bookTextData){
        this.setState({textForPages:bookTextData.text});
      }.bind(this))
      .catch(function(error){
        this.setState({textForPages:{0: "error retrieving book text"}});
      }.bind(this));
    }
  }

  componentDidMount(){
    this.slide();
  }

  slide(){
    this.refs.subPanel.slide();
  }

  render(){
      return (
        <SubPanel
          ref='subPanel'
          style={
            [this.props.style,
              {
                backgroundColor: '#1586BA'
              }
            ]
          } >
          <ScrollView>
            <Text style={{color:'white'}}>{this.state.textForPages[this.props.pageNumber]}</Text>
          </ScrollView>
        </SubPanel>
      );
  }
}

export default BookText;
