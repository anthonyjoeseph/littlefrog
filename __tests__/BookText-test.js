// __tests__/BookText-test.js
import 'react-native';
import React from 'react';
import BookText from '../app/scenes/main/BookReader/panels/BookText';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <BookText
      style={{
        position:'absolute',
        zIndex: 1,
        bottom: 0,
        height: 100,
        left: 0,
        width: 500
      }}
      pageNumber={ 0 } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
