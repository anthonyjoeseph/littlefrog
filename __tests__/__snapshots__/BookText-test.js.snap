exports[`test renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "backgroundColor": "#1586BA",
      "bottom": 0,
      "height": 100,
      "left": 0,
      "position": "absolute",
      "transform": Array [
        Object {
          "translateX": -1000,
        },
      ],
      "width": 500,
      "zIndex": 1,
    }
  }>
  <ScrollView>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }>
      Once upon a time, next to a pond, there was a village of frogs.
      “Ribbit, ribbit! Ribbit, ribbit!”
      The cries of the frogs made it a very noisy village!
      
    </Text>
  </ScrollView>
</View>
`;
